src/generators.py

```diff
@@ -0,0 +1,35 @@
+from typing import Dict, List, Union
+
+
+def filter_by_currency(transactions: List[Dict], currency: Union[str]) -> List[Dict]:
```

Comment:
Union используется, когда у тебя возможны несколько типов. Например строка или число определяются так `Union[str, int]`. В данном же случае только строка может быть:
```suggestion
def filter_by_currency(transactions: List[Dict], currency: str) -> List[Dict]:
```